jolly-jellyfish/src/page_maker/views.py

```diff
@@ -1,3 +1,219 @@
+    template_name = 'page_maker/webpage_detail.html'
+    context_object_name = 'webpage'
+    slug_field = 'name'
+    slug_url_kwarg = 'pagename'
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        webpage = self.get_object()
+        context['comments'] = Comment.objects.filter(parent_page=webpage)
+        context['comment_form'] = CommentForm()
+        return context
+
+
+class WebpageListView(ListView):
+    """
+    Display webpages ordered by vote number
+    """
+    model = Webpage
+    template = 'page_maker/webpage_list.html'
+    context_object_name = 'webpages'
+    paginate_by = 10
+    ordering = ['-votes']
+
+
+class WebpageUpdateView(LoginRequiredMixin, UserPassesTestMixin, UpdateView):
+    model = Webpage
+    template_name = 'page_maker/webpage_update.html'
+
+    def test_func(self):
+        webpage = self.get_object()
+        if webpage.author == self.request.user:
+            return True
+        return False
+
+    def get_queryset(self):
+        webpage_name = self.kwargs.get('pagename')
+        return Webpage.objects.filter(name=webpage_name)
+
+
+class WebpageDeleteView(LoginRequiredMixin, UserPassesTestMixin, DeleteView):
+    model = Webpage
+    template_name = 'page_maker/webpage_delete.html'
+
+    def test_func(self):
+        webpage = self.get_object()
+        if webpage.author == self.request.user or self.request.user.is_superuser:
+            return True
+        return False
+
+    def get_queryset(self):
+        webpage_name = self.kwargs.get('pagename')
+        return Webpage.objects.filter(name=webpage_name)
+
+
+class TemplateCreateView(LoginRequiredMixin, FormView):
+    template_name = 'page_maker/template_create.html'
+    form_class = TemplateForm
+
+    # TODO validate and sanitize style sheet
+    def form_valid(self, form):
+        self.form = form
+        form.instance.author = self.request.user
+        form.save()
+        return super().form_valid(form)
+
+    def get_success_url(self):
+        templatename = self.form.cleaned_data['name']
+        return reverse_lazy('template-view', kwargs={'templatename': templatename})
+
+
+class TemplateView(DetailView):
+    """
+    Display template using example page
+    """
+    model = Template
+    template_name = 'page_maker/example_page.html'
+    context_object_name = 'template'
+    slug_field = 'name'
+    slug_url_kwarg = 'templatename'
+
+
+class TemplateDeleteView(LoginRequiredMixin, UserPassesTestMixin, DeleteView):
+    model = Template
+    template_name = 'page_maker/template_delete.html'
+    slug_field = 'name'
+    slug_url_kwarg = 'templatename'
+
+    def test_func(self):
+        template = self.get_object()
+        if template.author == self.request.user or self.request.user.is_superuser:
+            return True
+        return False
+
+
+class CommentCreateView(LoginRequiredMixin, FormView):
+    http_method_names = ['post']  # POST request only
```

Comment:
Maybe capitalize `POST` in the method names? (Don't know how Django handles this...) But the comment is probably extraneous.

Reply:
Not sure; @baru64 ?